src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -439,22 +435,15 @@ private void InformAboutFilteredOutProjects()
             VsShellUtils.WriteToSonarLintOutputPane(this.host, output.ToString());
         }
 
-        private void InformAboutQualityProfileToDownload(string profileName, string profileKey, string languageName, bool isDownloadFailed)
+        private void InformAboutQualityProfileToDownload(QualityProfile profile, string languageName, bool isDownloadFailed)
         {
-            string output;
-
-            if (isDownloadFailed)
-            {
-                output = string.Format(Strings.QualityProfileDownloadFailedMessageFormat, profileName, profileKey, languageName);
-            }
-            else
-            {
-                output = string.Format(Strings.QualityProfileDownloadSuccessfulMessageFormat, profileName, profileKey, languageName);
-            }
-
-            output = string.Format(Strings.SubTextPaddingFormat, output);
-
-            VsShellUtils.WriteToSonarLintOutputPane(this.host, output);
+            var output =
+                profile == null
+                    ? string.Format(Strings.CannotDownloadQualityProfileForLanguage, languageName)
+                    : isDownloadFailed
+                        ? string.Format(Strings.QualityProfileDownloadFailedMessageFormat, profile.Name, profile.Key, languageName)
```

Comment:
You can merge these two `string.Format`s by assigning the format to a local variable.  Also, nested `?:`s are hard to read, let's put the second one in a dedicated method.